net/tlsdial/tlsdial.go

```diff
@@ -263,25 +264,32 @@ func SetConfigExpectedCertHash(c *tls.Config, wantFullCertSHA256Hex string) {
 		if len(rawCerts) == 0 {
 			return errors.New("no certs presented")
 		}
-		if len(rawCerts) > 1 {
-			return errors.New("unexpected multiple certs presented")
-		}
-		if fmt.Sprintf("%02x", sha256.Sum256(rawCerts[0])) != wantFullCertSHA256Hex {
-			return fmt.Errorf("cert hash does not match expected cert hash")
-		}
-		cert, err := x509.ParseCertificate(rawCerts[0])
-		if err != nil {
-			return fmt.Errorf("ParseCertificate: %w", err)
-		}
-		if err := cert.VerifyHostname(c.ServerName); err != nil {
-			return fmt.Errorf("cert does not match server name %q: %w", c.ServerName, err)
-		}
-		now := time.Now()
-		if now.After(cert.NotAfter) {
-			return fmt.Errorf("cert expired %v", cert.NotAfter)
-		}
-		if now.Before(cert.NotBefore) {
-			return fmt.Errorf("cert not yet valid until %v; is your clock correct?", cert.NotBefore)
+		numCerts := 0
+		for _, rawCert := range rawCerts {
+			cert, err := x509.ParseCertificate(rawCert)
+			if err != nil {
+				return fmt.Errorf("ParseCertificate: %w", err)
+			}
+			if strings.HasPrefix(cert.Subject.CommonName, "derpkey") {
+				continue
+			}
```

Comment:
Tying this special behavior back to derp server code via docs or constant would be helpful.

Reply:
done. ended up with a new package to not add silly deps.  (const packages are ~free)

Reply:
helps to actually `git add` the new file